File: frontend/src/membre.ts
import {ref} from 'vue'


interface Membre {
  id: number
  nom: string
  prenom: string
  email: string
  telephone: string
  cotisation_payee: boolean
  date_inscription: string
}
export const membres = ref<Membre[]>([]);
export type {Membre};

File: backend/Membre.py
from pydantic import BaseModel,ConfigDict, EmailStr
from datetime import datetime
from typing import Optional


class MembreCreate(BaseModel):
    """Modèle pour la création d'un membre"""
    nom: str
    prenom: str
    email: EmailStr
    telephone: str
    cotisation_payee: bool = False


class Membre(BaseModel):
    """Modèle complet d'un membre"""
    id: int
    nom: str
    prenom: str
    email: EmailStr
    telephone: str
    cotisation_payee: bool = False
    date_inscription: datetime

    class Config:
        from_attributes = True
